hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALSplitter.java

```diff
@@ -285,23 +286,35 @@ boolean splitLogFile(FileStatus logfile, CancelableProgressable reporter) throws
         String encodedRegionNameAsStr = Bytes.toString(region);
         lastFlushedSequenceId = lastFlushedSequenceIds.get(encodedRegionNameAsStr);
         if (lastFlushedSequenceId == null) {
-          if (sequenceIdChecker != null) {
-            RegionStoreSequenceIds ids = sequenceIdChecker.getLastSequenceId(region);
-            Map<byte[], Long> maxSeqIdInStores = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-            for (StoreSequenceId storeSeqId : ids.getStoreSequenceIdList()) {
-              maxSeqIdInStores.put(storeSeqId.getFamilyName().toByteArray(),
-                storeSeqId.getSequenceId());
+          if (!(isRegionDirPresentUnderRoot(entry.getKey().getTableName(), encodedRegionNameAsStr))) {
```

Comment:
Is this the right test to do? Should we rather ask the Master if this is an active Region? Asking the FS could be racy?  E.g. something like maser.getAM.getRegionStates().isRegionInRegionStates... It takes HRI which you might not have so you might need to add a version that takes encoded region name?

Reply:
Actually thought abt that and even I suggested as an option. Thought that will be bit more risky. Specially in case when the split/merge like ops happening. I feel race chances are more there. At FS level, when the region is created, we will have the region dir in place.

On asking, HM my idea was to use a special return value when we ask HM abt the max flushed seqId of the region. Now we return -1 when HM dont know this value for this region. so if we return -2 or so, we can treat it as invalid region.

But I felt that is more risky!

Reply:
Master is the authority on state. Reading the FS is reading tea leaves. Let me look at the patch again.